packages/react-scripts/config/webpack.config.dev.js

```diff
@@ -29,7 +29,7 @@ var publicPath = '/';
 // `publicUrl` is just like `publicPath`, but we will provide it to our app
 // as %PUBLIC_URL% in `index.html` and `process.env.PUBLIC_URL` in JavaScript.
 // Omit trailing slash as %PUBLIC_PATH%/xyz looks better than %PUBLIC_PATH%xyz.
-var publicUrl = '';
+var publicUrl = paths.servedPath.slice(0, -1);
```

Comment:
After reviewing that request with more attention, it seems that this only solves part of the use case. I think that user would desire the application served from the `/prefix`.
Since that's the case, I think it's best we revert this behavior and ask for a discussion issue to be opened on the topic. I think adding that behavior is out of the scope of this PR.